webapp/channels/src/components/widgets/loading/loading_spinner.tsx

```diff
@@ -12,26 +12,26 @@ type Props = {
     style?: React.CSSProperties;
 }
 
-export default class LoadingSpinner extends React.PureComponent<Props> {
-    public static defaultProps: Props = {
-        text: null,
-    };
+const LoadingSpinner: React.FC<Props> = React.memo(({text, style}) => {
+    return (
+        <span
+            id='loadingSpinner'
+            className={'LoadingSpinner' + (text ? ' with-text' : '')}
+            style={style}
+            data-testid='loadingSpinner'
+        >
+            <LocalizedIcon
+                className='fa fa-spinner fa-fw fa-pulse spinner'
+                component='span'
+                title={{id: t('generic_icons.loading'), defaultMessage: 'Loading Icon'}}
+            />
+            {text}
+        </span>
+    );
+});
 
-    public render() {
-        return (
-            <span
-                id='loadingSpinner'
-                className={'LoadingSpinner' + (this.props.text ? ' with-text' : '')}
-                style={this.props.style}
-                data-testid='loadingSpinner'
-            >
-                <LocalizedIcon
-                    className='fa fa-spinner fa-fw fa-pulse spinner'
-                    component='span'
-                    title={{id: t('generic_icons.loading'), defaultMessage: 'Loading Icon'}}
-                />
-                {this.props.text}
-            </span>
-        );
-    }
-}
+LoadingSpinner.defaultProps = {
```

Comment:
The `defaultProps` is not the recomended approach for functional components, is prefered to use the default value for parameters, something like: `const LoadingSpinner: React.FC<Props> = ({text = null, style}) => {`